narwhals/_pandas_like/utils.py

```diff
@@ -25,12 +25,15 @@
 from narwhals.exceptions import ShapeError
 
 if TYPE_CHECKING:
+    from collections.abc import Mapping
     from types import ModuleType
 
     from pandas._typing import Dtype as PandasDtype
     from pandas.core.dtypes.dtypes import BaseMaskedDtype
     from typing_extensions import TypeIs
 
+    from narwhals._arrow.typing import NativeIntervalUnit
```

Comment:
i think we should define `NativeIntervalUnit` here too, rather than reusing arrow's

Reply:
I added `NativeIntervalUnit` to the pandas-like utils